src/app/shared/case-editor/case-edit.component.ts

```diff
@@ -105,4 +104,13 @@ export class CaseEditComponent implements OnInit {
     this.confirmation = confirmation;
     return this.router.navigate(['confirm'], {relativeTo: this.route});
   }
+
+  private resetInvalidFormFields() {
+    Object.keys(this.form.controls['data'].value).forEach(key => {
+      const control = this.form.controls['data'].get(key);
+      if (control.touched && control.invalid) {
+        control.reset();
```

Comment:
Has this been tried for simple, complex and collection types?

Reply:
That's a good point. Haven't tested it for complex tipes yet. Will do cheers